File: src/modules/cliHandler.js
const { githubConnector, timestampProcessor, schedulerService, patternEngine } = require('./commitScheduler');
const sessionManager = require('./sessionManager');
const bannerUI = require('./bannerUI');
const aiHandler = require('./aiHandler');
const userConfigManager = require('./userConfigManager');

class CLIHandler {
  constructor() {
    this.commands = new Map();
    this.initializeCommands();
  }

  initializeCommands() {
    this.commands.set('showoptions', { handler: this.handleShowOptions.bind(this), description: 'Show the main banner with buttons' });
    this.commands.set('pastcommit', { handler: this.handlePastCommit.bind(this), description: 'Create a backdated commit' });
    this.commands.set('commitnow', { handler: this.handleCommitNow.bind(this), description: 'Create an immediate commit' });
    this.commands.set('schedule', { handler: this.handleSchedule.bind(this), description: 'Schedule recurring commits' });
    this.commands.set('pattern', { handler: this.handlePattern.bind(this), description: 'Generate pattern commits' });
    this.commands.set('cancel', { handler: this.handleCancel.bind(this), description: 'Cancel a scheduled job' });
    this.commands.set('commits', { handler: this.handleCommits.bind(this), description: 'List scheduled jobs' });
    this.commands.set('mystats', { handler: this.handleStats.bind(this), description: 'Show your session statistics' });
    this.commands.set('setrepo', { handler: this.handleSetRepo.bind(this), description: 'Configure your repository' });
    this.commands.set('detailed', { handler: this.handleDetailed.bind(this), description: 'Show detailed command information' });
    this.commands.set('help', { handler: this.handleHelp.bind(this), description: 'Show help information' });
    this.commands.set('start', { handler: this.handleStart.bind(this), description: 'Start a new session' });
    this.commands.set('ai-chat', { handler: this.handleAIChat.bind(this), description: 'Chat with AI assistant' });
    this.commands.set('setup', { handler: this.handleSetup.bind(this), description: 'Setup your GitHub credentials' });
    this.commands.set('mygithub', { handler: this.handleMyGitHub.bind(this), description: 'View your GitHub configuration' });
    this.commands.set('clearhistory', { handler: this.handleClearHistory.bind(this), description: 'Clear AI chat history' });
  }

  parseCommand(message) {
    const content = message.trim();
    
    if (!content.startsWith('!') && !content.startsWith('|')) return null;

    const isAI = content.startsWith('|');
    const prefix = isAI ? '|' : '!';
    const parts = content.slice(1).split(' ');
    const command = isAI ? 'ai-chat' : parts[0].toLowerCase();
    const args = parts.slice(1);

    return { command, args, fullArgs: parts.slice(1).join(' '), isAI };
  }

  async handle(message) {
    const parsed = this.parseCommand(message.content);
    if (!parsed) return null;

    const cmd = this.commands.get(parsed.command);
    if (!cmd) return null;

    try {
      return await cmd.handler(message, parsed.args, parsed);
    } catch (error) {
      console.error('CLI Command Error:', error);
      return bannerUI.createErrorEmbed('Command Error', error.message);
    }
  }

  async handleShowOptions(message) {
    const embed = bannerUI.createMainBanner(message.author.username);
    const buttons = bannerUI.createFeatureButtons();
    
    return {
      embeds: [embed],
      components: buttons
    };
  }

  async handlePastCommit(message, args) {
    if (args.length < 2) {
      return bannerUI.createErrorEmbed('Missing Arguments', 'Usage: !pastcommit <date> <time> "<message>"\nExample: !pastcommit 2024-01-15 14:30 "Fixed bug"');
    }

    const lastQuoteIndex = message.content.lastIndexOf('"');
    let msgArgs = [];
    
    for (let i = 0; i < args.length; i++) {
      if (args[i].startsWith('"') && args[i].length > 1) {
        const remaining = args.slice(i).join(' ');
        const endQuote = remaining.indexOf('"', 1);
        if (endQuote > -1) {
          msgArgs.push(remaining.slice(1, endQuote));
          break;
        }
      }
    }

    if (msgArgs.length === 0) {
      msgArgs = [args.slice(2).join(' ')];
    }

    const dateInput = args[0];
    const timeInput = args[1] && !args[1].includes('"') ? args[1] : null;
    const commitMessage = msgArgs[0] || args.slice(2).join(' ');

    if (!commitMessage) {
      return bannerUI.createErrorEmbed('Missing Message', 'Please provide a commit message');
    }

    let timestamp;
    try {
      const datePart = timeInput ? `${dateInput} ${timeInput}` : dateInput;
      const validation = timestampProcessor.validate(datePart);
      if (!validation.valid) {
        return bannerUI.createErrorEmbed('Invalid Date', validation.error);
      }
      timestamp = timestampProcessor.toISOString(validation.parsed);
    } catch (error) {
      return bannerUI.createErrorEmbed('Date Parsing Error', error.message);
    }

    const validation = githubConnector.validateConfig();
    if (!validation.valid) {
      return bannerUI.createErrorEmbed('Configuration Error', validation.errors.join('\n'));
    }

    try {
      await githubConnector.initialize();
    } catch (error) {
      return bannerUI.createErrorEmbed('Initialization Failed', error.message);
    }

    const result = await githubConnector.createCommit(commitMessage, timestamp);

    if (result.success) {
      const session = sessionManager.getActiveSession(message.author.id);
      if (session) {
        sessionManager.addCommitToSession(session.id, { message: commitMessage, type: 'backdated', success: true });
      }
      return bannerUI.createSuccessEmbed('Backdated Commit Created', `Successfully created commit: "${commitMessage}"`, [
        { name: '📅 Commit Date', value: new Date(timestamp).toLocaleString(), inline: true },
        { name: '🔢 Timestamp', value: timestamp, inline: true }
      ]);
    } else {
      return bannerUI.createErrorEmbed('Commit Failed', result.error);
    }
  }

  async handleCommitNow(message, args) {
    if (args.length === 0) {
      return bannerUI.createErrorEmbed('Missing Arguments', 'Usage: !commitnow "<message>"\nExample: !commitnow "Added new feature"');
    }

    const content = message.content.slice(message.content.indexOf(args[0]) - 1);
    const match = content.match(/"([^"]+)"/);
    const commitMessage = match ? match[1] : args.join(' ');

    const validation = githubConnector.validateConfig();
    if (!validation.valid) {
      return bannerUI.createErrorEmbed('Configuration Error', validation.errors.join('\n'));
    }

    try {
      await githubConnector.initialize();
    } catch (error) {
      return bannerUI.createErrorEmbed('Initialization Failed', error.message);
    }

    const result = await githubConnector.createCommit(commitMessage, new Date().toISOString());

    if (result.success) {
      const session = sessionManager.getActiveSession(message.author.id);
      if (session) {
        sessionManager.addCommitToSession(session.id, { message: commitMessage, type: 'normal', success: true });
      }
      return bannerUI.createSuccessEmbed('Commit Created', `Successfully created commit: "${commitMessage}"`, [
        { name: '📅 Timestamp', value: new Date().toISOString(), inline: true },
        { name: '📤 Pushed', value: result.pushed ? 'Yes' : 'No', inline: true }
      ]);
    } else {
      return bannerUI.createErrorEmbed('Commit Failed', result.error);
    }
  }

  async handleSchedule(message, args) {
    if (args.length < 2) {
      return bannerUI.createErrorEmbed('Missing Arguments', 'Usage: !schedule "<cron>" "<message>"\nExample: !schedule "0 9 * * 1" "Weekly report"');
    }

    const cronMatch = message.content.match(/"([^"]+)"/g);
    if (!cronMatch || cronMatch.length < 2) {
      return bannerUI.createErrorEmbed('Format Error', 'Use quotes for cron expression and message\nExample: !schedule "0 9 * * 1" "Weekly report"');
    }

    const cronExpr = cronMatch[0].replace(/"/g, '');
    const commitMessage = cronMatch[1].replace(/"/g, '');

    if (!schedulerService.validateCronExpression(cronExpr)) {
      return bannerUI.createErrorEmbed('Invalid Cron', 'Please provide a valid cron expression\nExample: 0 9 * * 1 (every Monday at 9am)');
    }

    const result = schedulerService.createJob(message.author.id, cronExpr, commitMessage);

    if (result.success) {
      const session = sessionManager.getActiveSession(message.author.id);
      if (session) {
        sessionManager.addCommitToSession(session.id, { message: commitMessage, type: 'scheduled', success: true });
      }
      return bannerUI.createSuccessEmbed('Schedule Created', `Scheduled commit: "${commitMessage}"`, [
        { name: '🆔 Job ID', value: result.job.id, inline: true },
        { name: '⏰ Schedule', value: cronExpr, inline: true },
        { name: '📊 Status', value: result.job.status, inline: true }
      ]);
    } else {
      return bannerUI.createErrorEmbed('Schedule Failed', result.error);
    }
  }

  async handlePattern(message, args) {
    if (args.length < 2) {
      return bannerUI.createErrorEmbed('Missing Arguments', 'Usage: !pattern <template> "<message>"\nExample: !pattern heart "contribution"');
    }

    const template = args[0].toLowerCase();
    const content = message.content.slice(message.content.indexOf(args[1]) - 1);
    const match = content.match(/"([^"]+)"/);
    const commitMessage = match ? match[1] : args.slice(1).join(' ');

    const preview = patternEngine.previewPattern(template);
    if (!preview.success) {
      return bannerUI.createErrorEmbed('Template Error', preview.error + '\nAvailable: A, B, C, heart, star, check, smiley');
    }

    const validation = githubConnector.validateConfig();
    if (!validation.valid) {
      return bannerUI.createErrorEmbed('Configuration Error', validation.errors.join('\n'));
    }

    try {
      await githubConnector.initialize();
    } catch (error) {
      return bannerUI.createErrorEmbed('Initialization Failed', error.message);
    }

    const result = await patternEngine.executePattern(template, commitMessage, { push: true, delayBetweenCommits: 1000 });

    if (result.success) {
      const session = sessionManager.getActiveSession(message.author.id);
      if (session) {
        sessionManager.addCommitToSession(session.id, { message: `Pattern: ${template}`, type: 'pattern', success: true });
      }
      return bannerUI.createSuccessEmbed('Pattern Commits Complete', `Created ${result.succeeded} commits using ${template} pattern`, [
        { name: '📊 Summary', value: `Succeeded: ${result.succeeded}, Failed: ${result.failed}`, inline: true }
      ]);
    } else {
      return bannerUI.createErrorEmbed('Pattern Failed', `Succeeded: ${result.succeeded}, Failed: ${result.failed}`);
    }
  }

  async handleCancel(message, args) {
    if (args.length === 0) {
      return bannerUI.createErrorEmbed('Missing Arguments', 'Usage: !cancel <job-id>\nUse !commits to see job IDs');
    }

    const jobId = args[0];
    const job = schedulerService.getJob(jobId);

    if (!job) {
      return bannerUI.createErrorEmbed('Job Not Found', `No scheduled job found with ID: ${jobId}`);
    }

    if (job.ownerId !== message.author.id) {
      return bannerUI.createErrorEmbed('Permission Denied', 'You can only cancel jobs you created');
    }

    const result = schedulerService.cancelJob(jobId);

    if (result.success) {
      return bannerUI.createSuccessEmbed('Schedule Cancelled', `Job ${jobId} has been stopped`, [
        { name: 'Message', value: job.message, inline: true },
        { name: 'Schedule', value: job.cronExpression, inline: true }
      ]);
    } else {
      return bannerUI.createErrorEmbed('Cancel Failed', result.error);
    }
  }

  async handleCommits(message, args) {
    const jobs = schedulerService.listJobs({ ownerId: message.author.id, limit: 10 });

    if (jobs.length === 0) {
      return bannerUI.createInfoEmbed('No Scheduled Jobs', 'You have no scheduled commit jobs. Use !schedule to create one!');
    }

    const embed = new EmbedBuilder()
      .setTitle('📋 Your Scheduled Jobs')
      .setDescription(`Showing ${jobs.length} job(s)`)
      .setColor(0x0099ff)
      .setTimestamp();

    jobs.forEach(job => {
      embed.addFields({
        name: `🆔 ${job.id.substring(0, 8)}...`,
        value: `**Message:** ${job.message}\n**Schedule:** \`${job.cronExpression}\`\n**Status:** ${job.status}\n**Next:** ${job.nextRun ? new Date(job.nextRun).toLocaleString() : 'N/A'}`,
        inline: false
      });
    });

    return embed;
  }

  async handleStats(message) {
    const stats = sessionManager.getSessionStats(message.author.id);

    return new EmbedBuilder()
      .setTitle('📊 Your Session Statistics')
      .setDescription('Statistics across all your sessions (last 30 days)')
      .setColor(0x00ccff)
      .addFields(
        { name: '📅 Total Sessions', value: stats.totalSessions.toString(), inline: true },
        { name: '✅ Active Sessions', value: stats.activeSessions.toString(), inline: true },
        { name: '📝 Total Commits', value: stats.totalCommits.toString(), inline: true },
        { name: '⏰ Backdated Commits', value: stats.backdatedCommits.toString(), inline: true },
        { name: '📅 Scheduled Commits', value: stats.scheduledCommits.toString(), inline: true },
        { name: '🎨 Pattern Commits', value: stats.patternCommits.toString(), inline: true },
        { name: '🕐 Oldest Session', value: stats.oldestSession ? new Date(stats.oldestSession).toLocaleDateString() : 'N/A', inline: true },
        { name: '🆕 Newest Session', value: stats.newestSession ? new Date(stats.newestSession).toLocaleDateString() : 'N/A', inline: true }
      )
      .setTimestamp();
  }

  async handleSetRepo(message, args) {
    if (args.length === 0) {
      return bannerUI.createErrorEmbed('Missing Arguments', 'Usage: !setrepo <path> or !setrepo github <owner/repo>\nExample: !setrepo ./myrepo\nExample: !setrepo github username/repo');
    }

    if (args[0].toLowerCase() === 'github' && args[1]) {
      const repoPath = args[1];
      githubConnector.configure({
        repoPath: `./${repoPath.replace('/', '-')}`,
        branch: args[2] || 'main',
        remote: 'origin'
      });
      return bannerUI.createSuccessEmbed('Repository Configured', `GitHub repository set to: ${repoPath}`, [
        { name: '📁 Repository', value: repoPath, inline: true },
        { name: '🌿 Branch', value: args[2] || 'main', inline: true }
      ]);
    } else {
      const repoPath = args[0];
      githubConnector.configure({
        repoPath: repoPath,
        branch: args[1] || 'main',
        remote: 'origin'
      });
      return bannerUI.createSuccessEmbed('Repository Configured', `Local repository set to: ${repoPath}`, [
        { name: '📁 Path', value: repoPath, inline: true },
        { name: '🌿 Branch', value: args[1] || 'main', inline: true }
      ]);
    }
  }

  async handleDetailed(message) {
    return bannerUI.createDetailedHelpEmbed();
  }

  async handleHelp(message) {
    return bannerUI.createFeaturesListEmbed();
  }

  async handleStart(message) {
    let session = sessionManager.getActiveSession(message.author.id);
    
    if (session) {
      return bannerUI.createInfoEmbed('Session Active', `You already have an active session started at ${new Date(session.startTime).toLocaleString()}`);
    }

    const config = githubConnector.config;
    session = sessionManager.createSession(message.author.id, message.author.username, {
      repoPath: config.repoPath,
      repoRemote: config.remote,
      branch: config.branch
    });

    return bannerUI.createSuccessEmbed('🎉 Session Started!', `Welcome ${message.author.username}! Your session has been created.`, [
      { name: '🆔 Session ID', value: session.id, inline: true },
      { name: '📅 Started At', value: new Date(session.startTime).toLocaleString(), inline: true },
      { name: '⏰ Duration', value: '30 days (auto-expires)', inline: true }
    ]);
  }

  async handleAIChat(message, args, parsed) {
    if (!aiHandler.isEnabled()) {
      return aiHandler.createAIBadResponseEmbed('AI is not enabled. Please set AI_ENABLED=true in .env');
    }

    const userMessage = parsed.fullArgs || '';
    
    if (!userMessage.trim()) {
      const embed = aiHandler.createAIBanner();
      return embed;
    }

    const userId = message.author.id;
    const result = await aiHandler.chat(userId, userMessage);

    if (result.success) {
      const chunks = this.splitMessage(result.response);
      
      if (chunks.length === 1) {
        return new EmbedBuilder()
          .setTitle('🤖 AI Response')
          .setDescription(result.response)
          .setColor(0x9b59b6)
          .setTimestamp();
      }
      
      const embeds = chunks.map((chunk, index) => 
        new EmbedBuilder()
          .setTitle(index === 0 ? '🤖 AI Response' : '📄 (continued)')
          .setDescription(chunk)
          .setColor(0x9b59b6)
          .setTimestamp()
      );
      
      return embeds;
    } else {
      return aiHandler.createAIBadResponseEmbed(result.error);
    }
  }

  splitMessage(text, maxLength = 2000) {
    const chunks = [];
    if (text.length <= maxLength) {
      return [text];
    }
    
    const parts = text.split('\n');
    let current = '';
    
    for (const part of parts) {
      if ((current + '\n' + part).length <= maxLength) {
        current += (current ? '\n' : '') + part;
      } else {
        if (current) chunks.push(current);
        if (part.length > maxLength) {
          const subParts = [];
          while (part.length > maxLength) {
            subParts.push(part.slice(0, maxLength));
            part = part.slice(maxLength);
          }
          current = part;
          chunks.push(...subParts.slice(0, -1));
        } else {
          current = part;
        }
      }
    }
    
    if (current) chunks.push(current);
    return chunks;
  }

  async handleSetup(message, args) {
    if (args.length < 2) {
      return bannerUI.createErrorEmbed('Missing Arguments', 
        'Usage: !setup github <github-pat> <repo-url> [branch]\n' +
        'Example: !setup github ghp_xxx https://github.com/user/repo main'
      );
    }

    if (args[0].toLowerCase() !== 'github') {
      return bannerUI.createErrorEmbed('Invalid Setup', 'Currently only GitHub setup is supported\nUse: !setup github <pat> <repo-url> [branch]');
    }

    const pat = args[1];
    const repoUrl = args[2];
    const branch = args[3] || 'main';

    const userId = message.author.id;
    let config = userConfigManager.getUserConfig(userId);
    
    if (!config) {
      config = userConfigManager.createUserConfig(userId, message.author.username);
    }

    const result = userConfigManager.setRepoConfig(userId, repoUrl, branch, 'origin');
    if (!result.success) {
      return bannerUI.createErrorEmbed('Setup Failed', result.error);
    }

    userConfigManager.setGitHubPAT(userId, pat);

    return bannerUI.createSuccessEmbed('✅ GitHub Setup Complete!', 
      `Your GitHub credentials have been saved securely.`, [
        { name: '📁 Repository', value: repoUrl, inline: true },
        { name: '🌿 Branch', value: branch, inline: true },
        { name: '🔐 Status', value: 'Credentials saved', inline: true }
      ]
    );
  }

  async handleMyGitHub(message) {
    const userId = message.author.id;
    const config = userConfigManager.getUserConfig(userId);

    if (!config) {
      return bannerUI.createInfoEmbed('No Setup Found', 
        'You haven\'t configured your GitHub yet.\nUse `!setup github <pat> <repo-url>` to set up.'
      );
    }

    const hasSetup = userConfigManager.hasGitHubSetup(userId);

    return new EmbedBuilder()
      .setTitle('📡 Your GitHub Configuration')
      .setColor(hasSetup ? 0x00ff00 : 0xffaa00)
      .addFields(
        { name: '🔐 PAT Configured', value: hasSetup ? '✅ Yes' : '❌ No', inline: true },
        { name: '📁 Repository', value: config.github?.repoPath || 'Not set', inline: true },
        { name: '🌿 Branch', value: config.github?.branch || 'main', inline: true },
        { name: '🔗 Remote', value: config.github?.remote || 'origin', inline: true },
        { name: '📅 Last Updated', value: new Date(config.updatedAt).toLocaleString(), inline: true }
      )
      .setTimestamp();
  }

  async handleClearHistory(message) {
    const userId = message.author.id;
    aiHandler.clearHistory(userId);
    
    return bannerUI.createSuccessEmbed('🗑️ Chat History Cleared', 
      'Your AI conversation history has been reset.'
    );
  }
}

const { EmbedBuilder } = require('discord.js');
module.exports = new CLIHandler();

File: BOT/python-bot/modules/cli_handler.py
import re
import discord

class CLIHandler:
    def __init__(self, ai_handler, user_config_manager, session_manager, banner_ui):
        self.ai_handler = ai_handler
        self.user_config_manager = user_config_manager
        self.session_manager = session_manager
        self.banner_ui = banner_ui
        
        self.commands = {
            'showoptions': self.handle_show_options,
            'pastcommit': self.handle_past_commit,
            'commitnow': self.handle_commit_now,
            'schedule': self.handle_schedule,
            'pattern': self.handle_pattern,
            'cancel': self.handle_cancel,
            'commits': self.handle_commits,
            'mystats': self.handle_stats,
            'setrepo': self.handle_set_repo,
            'detailed': self.handle_detailed,
            'help': self.handle_help,
            'start': self.handle_start,
            'setup': self.handle_setup,
            'mygithub': self.handle_my_github,
            'clearhistory': self.handle_clear_history,
            'adminjobs': self.handle_admin_jobs,
        }

    def parse_command(self, content):
        content = content.strip()
        
        if not content.startswith('!') and not content.startswith('|'):
            return None

        is_ai = content.startswith('|')
        prefix = '|' if is_ai else '!'
        parts = content[len(prefix):].split()
        command = 'ai-chat' if is_ai else parts[0].lower()
        args = parts[1:]

        return {
            'command': command,
            'args': args,
            'full_args': ' '.join(args),
            'is_ai': is_ai
        }

    async def handle(self, message):
        parsed = self.parse_command(message.content)
        if not parsed:
            return None

        cmd = self.commands.get(parsed['command'])
        if not cmd:
            return None

        try:
            return await cmd(message, parsed['args'], parsed)
        except Exception as e:
            return self.banner_ui.create_error_embed('Command Error', str(e))

    async def handle_show_options(self, message, args, parsed):
        embed = self.banner_ui.create_main_banner(message.author.name)
        view = self.banner_ui.create_feature_buttons()
        return embed

    async def handle_setup(self, message, args, parsed):
        if len(args) < 3:
            return self.banner_ui.create_error_embed(
                'Missing Arguments',
                'Usage: !setup github <github-pat> <repo-url> [branch]\n'
                'Example: !setup github ghp_xxx https://github.com/user/repo main'
            )

        if args[0].lower() != 'github':
            return self.banner_ui.create_error_embed(
                'Invalid Setup',
                'Currently only GitHub setup is supported\nUse: !setup github <pat> <repo-url> [branch]'
            )

        pat = args[1]
        repo_url = args[2]
        branch = args[3] if len(args) > 3 else 'main'
        user_id = str(message.author.id)
        
        config = self.user_config_manager.get_user_config(user_id)
        if not config:
            config = self.user_config_manager.create_user_config(user_id, message.author.name)

        result = self.user_config_manager.set_repo_config(user_id, repo_url, branch, 'origin')
        if not result['success']:
            return self.banner_ui.create_error_embed('Setup Failed', result['error'])

        return self.banner_ui.create_success_embed(
            '✅ GitHub Setup Complete!',
            'Your GitHub credentials have been saved securely.',
            [
                {'name': '📁 Repository', 'value': repo_url, 'inline': True},
                {'name': '🌿 Branch', 'value': branch, 'inline': True},
                {'name': '🔐 Status', 'value': 'Credentials saved', 'inline': True}
            ]
        )

    async def handle_my_github(self, message, args, parsed):
        user_id = str(message.author.id)
        config = self.user_config_manager.get_user_config(user_id)

        if not config:
            return self.banner_ui.create_info_embed(
                'No Setup Found',
                'You haven\'t configured your GitHub yet.\nUse `!setup github <pat> <repo-url>` to set up.'
            )

        has_setup = self.user_config_manager.has_github_setup(user_id)

        embed = discord.Embed(
            title='📡 Your GitHub Configuration',
            color=0x00ff00 if has_setup else 0xffaa00
        )
        embed.add_field(name='🔐 PAT Configured', value='✅ Yes' if has_setup else '❌ No', inline=True)
        embed.add_field(name='📁 Repository', value=config.get('github', {}).get('repo_path') or 'Not set', inline=True)
        embed.add_field(name='🌿 Branch', value=config.get('github', {}).get('branch') or 'main', inline=True)
        embed.add_field(name='🔗 Remote', value=config.get('github', {}).get('remote') or 'origin', inline=True)
        embed.add_field(name='📅 Last Updated', value=config.get('updated_at', 'N/A'), inline=True)
        return embed

    async def handle_clear_history(self, message, args, parsed):
        user_id = str(message.author.id)
        self.ai_handler.clear_history(user_id)
        
        return self.banner_ui.create_success_embed(
            '🗑️ Chat History Cleared',
            'Your AI conversation history has been reset.'
        )

    async def handle_admin_jobs(self, message, args, parsed):
        user_id = str(message.author.id)
        
        is_admin = message.author.guild_permissions.administrator if hasattr(message.author, 'guild') else False
        
        if not is_admin:
            return self.banner_ui.create_error_embed(
                'Admin Only',
                'Only administrators can view all scheduled jobs.'
            )

        from modules.scheduler_service import SchedulerService
        jobs = SchedulerService.list_all_jobs()

        if not jobs:
            return self.banner_ui.create_info_embed('No Scheduled Jobs', 'There are no scheduled jobs.')

        embed = discord.Embed(
            title='📋 All Scheduled Jobs',
            description=f'Total: {len(jobs)} job(s)',
            color=0x0099ff
        )

        for job in jobs[:10]:
            embed.add_field(
                name=f"🆔 {job.get('id', 'N/A')[:8]}...",
                value=f"**Owner:** {job.get('owner_id', 'N/A')[:8]}...\n"
                      f"**Message:** {job.get('message', 'N/A')}\n"
                      f"**Schedule:** `{job.get('cron_expression', 'N/A')}`\n"
                      f"**Status:** {job.get('status', 'N/A')}",
                inline=False
            )

        return embed

    async def handle_ai_chat(self, message, args, parsed):
        if not self.ai_handler.is_enabled():
            return self.ai_handler.create_error_embed(
                'AI is not enabled. Please set AI_ENABLED=true in .env'
            )

        user_message = parsed['full_args']
        
        if not user_message.strip():
            return self.ai_handler.create_ai_banner()

        user_id = str(message.author.id)
        result = await self.ai_handler.chat(user_id, user_message)

        if result['success']:
            chunks = self.split_message(result['response'])
            
            if len(chunks) == 1:
                return discord.Embed(
                    title='🤖 AI Response',
                    description=result['response'],
                    color=0x9b59b6
                )
            
            embeds = []
            for i, chunk in enumerate(chunks):
                embeds.append(discord.Embed(
                    title='🤖 AI Response' if i == 0 else '📄 (continued)',
                    description=chunk,
                    color=0x9b59b6
                ))
            return embeds
        elif result.get('rate_limited'):
            return self.ai_handler.create_rate_limit_embed(
                result['error'],
                result.get('wait_seconds', 60)
            )
        else:
            return self.ai_handler.create_error_embed(result['error'])

    def split_message(self, text, max_length=2000):
        if len(text) <= max_length:
            return [text]
        
        chunks = []
        parts = text.split('\n')
        current = ''
        
        for part in parts:
            if len(current + '\n' + part) <= max_length:
                current += ('\n' if current else '') + part
            else:
                if current:
                    chunks.append(current)
                if len(part) > max_length:
                    while len(part) > max_length:
                        chunks.append(part[:max_length])
                        part = part[max_length:]
                    current = part
                else:
                    current = part
        
        if current:
            chunks.append(current)
        
        return chunks if chunks else [text]

    async def handle_start(self, message, args, parsed):
        user_id = str(message.author.id)
        session = self.session_manager.get_active_session(user_id)
        
        if session:
            return self.banner_ui.create_info_embed(
                'Session Active',
                f"You already have an active session started at {session.get('start_time', 'N/A')}"
            )

        session = self.session_manager.create_session(user_id, message.author.name, {
            'repo_path': os.getenv('TARGET_REPO_PATH', './repo'),
            'repo_remote': os.getenv('TARGET_REPO_REMOTE', 'origin'),
            'branch': os.getenv('TARGET_REPO_BRANCH', 'main')
        })

        return self.banner_ui.create_success_embed(
            '🎉 Session Started!',
            f"Welcome {message.author.name}! Your session has been created.",
            [
                {'name': '🆔 Session ID', 'value': session.get('id', 'N/A'), 'inline': True},
                {'name': '📅 Started At', 'value': session.get('start_time', 'N/A'), 'inline': True},
                {'name': '⏰ Duration', 'value': '30 days (auto-expires)', 'inline': True}
            ]
        )

    async def handle_stats(self, message, args, parsed):
        user_id = str(message.author.id)
        stats = self.session_manager.get_session_stats(user_id)

        embed = discord.Embed(
            title='📊 Your Session Statistics',
            description='Statistics across all your sessions (last 30 days)',
            color=0x00ccff
        )
        embed.add_field(name='📅 Total Sessions', value=str(stats.get('total_sessions', 0)), inline=True)
        embed.add_field(name='✅ Active Sessions', value=str(stats.get('active_sessions', 0)), inline=True)
        embed.add_field(name='📝 Total Commits', value=str(stats.get('total_commits', 0)), inline=True)
        return embed

    async def handle_help(self, message, args, parsed):
        return self.banner_ui.create_features_list_embed()

    async def handle_detailed(self, message, args, parsed):
        return self.banner_ui.create_detailed_help_embed()

    async def handle_commits(self, message, args, parsed):
        user_id = str(message.author.id)
        from modules.scheduler_service import SchedulerService
        jobs = SchedulerService.list_jobs(owner_id=user_id, limit=10)

        if not jobs:
            return self.banner_ui.create_info_embed(
                'No Scheduled Jobs',
                'You have no scheduled commit jobs. Use !schedule to create one!'
            )

        embed = discord.Embed(
            title='📋 Your Scheduled Jobs',
            description=f'Showing {len(jobs)} job(s)',
            color=0x0099ff
        )

        for job in jobs:
            embed.add_field(
                name=f"🆔 {job.get('id', 'N/A')[:8]}...",
                value=f"**Message:** {job.get('message', 'N/A')}\n"
                      f"**Schedule:** `{job.get('cron_expression', 'N/A')}`\n"
                      f"**Status:** {job.get('status', 'N/A')}",
                inline=False
            )

        return embed

    async def handle_cancel(self, message, args, parsed):
        if not args:
            return self.banner_ui.create_error_embed(
                'Missing Arguments',
                'Usage: !cancel <job-id>\nUse !commits to see job IDs'
            )

        job_id = args[0]
        from modules.scheduler_service import SchedulerService
        job = SchedulerService.get_job(job_id)

        if not job:
            return self.banner_ui.create_error_embed('Job Not Found', f'No scheduled job found with ID: {job_id}')

        if job.get('owner_id') != str(message.author.id):
            return self.banner_ui.create_error_embed('Permission Denied', 'You can only cancel jobs you created')

        result = SchedulerService.cancel_job(job_id)

        if result.get('success'):
            return self.banner_ui.create_success_embed(
                'Schedule Cancelled',
                f'Job {job_id} has been stopped',
                [
                    {'name': 'Message', 'value': job.get('message', 'N/A'), 'inline': True},
                    {'name': 'Schedule', 'value': job.get('cron_expression', 'N/A'), 'inline': True}
                ]
            )
        else:
            return self.banner_ui.create_error_embed('Cancel Failed', result.get('error'))

    async def handle_past_commit(self, message, args, parsed):
        return self.banner_ui.create_info_embed('Coming Soon', 'Backdated commits coming soon!')

    async def handle_commit_now(self, message, args, parsed):
        return self.banner_ui.create_info_embed('Coming Soon', 'Commit now coming soon!')

    async def handle_schedule(self, message, args, parsed):
        return self.banner_ui.create_info_embed('Coming Soon', 'Scheduling coming soon!')

    async def handle_pattern(self, message, args, parsed):
        return self.banner_ui.create_info_embed('Coming Soon', 'Pattern commits coming soon!')

    async def handle_set_repo(self, message, args, parsed):
        return self.banner_ui.create_info_embed('Coming Soon', 'Set repo coming soon!')

import os
